uniro_frontend/src/component/NavgationMap.tsx

```diff
@@ -1,109 +1,182 @@
 import { useEffect, useRef } from "react";
 import useMap from "../hooks/useMap";
-import { NavigationRoute } from "../data/types/route";
+import { CautionRoute, DangerRoute, NavigationRouteList } from "../data/types/route";
 import createAdvancedMarker from "../utils/markers/createAdvanedMarker";
 import createMarkerElement from "../components/map/mapMarkers";
 import { Markers } from "../constant/enum/markerEnum";
-import useSuspenseMap from "../hooks/useSuspenseMap";
+import { Coord } from "../data/types/coord";
+import useRoutePoint from "../hooks/useRoutePoint";
+import { AdvancedMarker } from "../data/types/marker";
 
 type MapProps = {
 	style?: React.CSSProperties;
-	routes: NavigationRoute;
-	/** 바텀시트나 상단 UI에 의해 가려지는 영역이 있을 경우, 지도 fitBounds에 추가할 패딩값 */
+	routeResult: NavigationRouteList;
+	risks: {
+		dangerRoutes: DangerRoute[];
+		cautionRoutes: CautionRoute[];
+	};
+	isDetailView: boolean;
 	topPadding?: number;
 	bottomPadding?: number;
 };
 
 // TODO: useEffect로 경로가 모두 로딩된 이후에 마커가 생성되도록 수정하기
 // TODO: 경로 로딩 완료시 살짝 zoomIn 하는 부분 구현하기
 
-const NavigationMap = ({ style, routes, topPadding = 0, bottomPadding = 0 }: MapProps) => {
-	const { mapRef, map, AdvancedMarker, Polyline } = useSuspenseMap();
+const NavigationMap = ({ style, routeResult, risks, isDetailView, topPadding = 0, bottomPadding = 0 }: MapProps) => {
+	const { mapRef, map, AdvancedMarker, Polyline } = useMap();
+	const { origin, destination } = useRoutePoint();
 
 	const boundsRef = useRef<google.maps.LatLngBounds | null>(null);
+	const markersRef = useRef<AdvancedMarker[]>([]);
 
 	if (!style) {
 		style = { height: "100%", width: "100%" };
 	}
 
 	useEffect(() => {
-		if (!mapRef || !map || !AdvancedMarker || !routes || !Polyline) return;
+		if (!mapRef || !map || !AdvancedMarker || !routeResult || !Polyline) return;
+
+		if (routeResult.routes.length === 0) return;
+
+		const paths: Coord[] = [];
+		const cautionFactor: Coord[] = [];
+
+		const { routes, routeDetails } = routeResult;
+
+		// 하나의 길 완성
+		paths.push(routes[0].node1);
+		routes.forEach((route) => {
+			const { node2 } = route;
+			paths.push(node2);
+		});
```

Comment:
```suggestion
    const paths = [...routes[0].node1, routes.map(el => el.node2)];
```

한 줄로 요약할 수 있을 것 같습니다.

Reply:
좋은 피드백 감사합니다!